File: src/backend/Subol.Core/Entities/Order.cs
namespace Subol.Core.Entities;

public class Order
{
    public int Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public int CustomerId { get; set; }
    public string? LocationId { get; set; }
    public string? DistributorId { get; set; }
    public decimal Cost { get; set; }
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ConfirmedAt { get; set; }
    
    public Customer Customer { get; set; } = null!;
    public User? Distributor { get; set; }
}

File: src/types/order.ts
import { Customer, Location } from "./customer";
import { Distributor } from "./distributor";

export interface OrderRequest {
  id?: number;
  orderNumber?: string;
  customerId?: number;
  locationId?: number;
  cost: number | undefined | null;
  statusString: 'New' | 'Pending' | 'Confirmed' | 'Draft';
  distributorId?: string;
}

export interface DistributorInfo {
  id: string;
  name: string;
  phone: string;
}

export interface OrderList {
  id: number;
  orderNumber: string;
  customer: Customer;
  location: Location;
  distributor: DistributorInfo;
  cost: number | undefined | null;
  status: string;
  createdAt: string;
  confirmedAt: string;
}

export interface PaginatedOrders {
  items: OrderList[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface OrderFilters {
  distributorId?: string | null;
  status?: string | null;
  page?: number;
  pageSize?: number;
  dateFrom?: string;
  dateTo?: string;
}